app/Services/UserService.php

```diff
@@ -17,4 +17,16 @@ public function createNewUser(array $params) {
             'password' => Hash::make($params['password'])
         ]);
     }
+
+    public function updateName(array $params) {
+        $user = auth()->user();
+        $user->name = $params['name'];
+        $user->save();
+    }
+
+    public function updatePassword(array $params) {
+        $user = auth()->user();
+        $user->password = Hash::make($params['password']);
+        $user->save();
+    }
```

Comment:
These require that auth() is defined. For example if we were to make a CLI command to use this service, it couldn't call these functions. Let's instead pass in the User we want to update as an argument.